### PR TITLE
git: Decommission libgit2 implementation

### LINKED ISSUE #478: figure out the future of package `git/libgit2`
we currently have two git implementations in the form of packages `git/gogit` and `git/libgit2`. the most notable difference between them is that `git/libgit2` uses CGO. due to this, we have several hacks in our project Makefile on top of the already messy build system we have for this particular package. imo, there are three options we can choose form:
* do nothing. keep things as it is, i.e. no features will be added and support will be bare minimum.
* remove it from existence.
* move it to a different repository and archive the repo to explicitly imply that the package is not maintained.

i can't think of other options, but more options are obviously welcome.

### PR DESCRIPTION
**Pull request by stefanprodan**
Decommission the libgit2 package as it is no longer used in the Flux project.

Closes #478

### CODE REVIEW

**Review by hiddeco — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -29,31 +29,17 @@ fmt-%:
...
 generate-%: controller-gen
-# Run schemapatch to validate all the kubebuilder markers before generation
-	@if [ "$(PKG)" = "git:libgit2" ]; then \
-		cd $(subst :,/,$*); make generate ;\
-	else \
-		cd $(subst :,/,$*); CGO_ENABLED=1 $(CONTROLLER_GEN) schemapatch:manifests="./" paths="./..." ;\
-		CGO_ENABLED=1 $(CONTROLLER_GEN) object:headerFile="$(root_dir)/hack/boilerplate.go.txt" paths="./..." ;\
-	fi
+	cd $(subst :,/,$*); CGO_ENABLED=1 $(CONTROLLER_GEN) schemapatch:manifests="./" paths="./..." ;\
+	CGO_ENABLED=1 $(CONTROLLER_GEN) object:headerFile="$(root_dir)/hack/boilerplate.go.txt" paths="./..." ;\
```
Comment (aryan9600) [inline Makefile:37]:
```suggestion
	cd $(subst :,/,$*); CGO_ENABLED=0 $(CONTROLLER_GEN) schemapatch:manifests="./" paths="./..." ;\
	CGO_ENABLED=0 $(CONTROLLER_GEN) object:headerFile="$(root_dir)/hack/boilerplate.go.txt" paths="./..." ;\
```

File: git/internal/e2e/run.sh [hunk 2]
```diff
@@ -20,6 +17,4 @@ if [[ "${GO_TEST_PREFIX}" = "" ]] || [[ "${GO_TEST_PREFIX}" = *"TestGitLabCEE2E"
 fi
 
 cd "${DIR}"
-CGO_LDFLAGS=$(PKG_CONFIG_PATH="${PKG_CONFIG_PATH}" pkg-config --libs --static --cflags libgit2)
-PKG_CONFIG_PATH="${PKG_CONFIG_PATH}" CGO_LDFLAGS="${CGO_LDFLAGS}" CGO_ENABLED=1 \
-    go test -v -tags 'netgo,osusergo,static_build,e2e' -race -run "^${GO_TEST_PREFIX}.*" ./...
+CGO_ENABLED=1 go test -v -tags 'netgo,osusergo,static_build,e2e' -race -run "^${GO_TEST_PREFIX}.*" ./...
```
Comment (aryan9600) [inline git/internal/e2e/run.sh:20]:
```suggestion
CGO_ENABLED=0 go test -v -tags 'netgo,osusergo,static_build,e2e' -race -run "^${GO_TEST_PREFIX}.*" ./...
```

File: git/libgit2/README.md [hunk 3]
```diff
@@ -0,0 +1,6 @@
+# Decommission notice
+
+The `github.com/fluxcd/pkg/git/libgit2` has been decommissioned as it is no longer used in the Flux project.
+
+The last package version that ships with libgit2 is
+[git/v0.10.0](https://github.com/fluxcd/pkg/releases/tag/git%2Fv0.10.0).
```
Comment (aryan9600) [inline git/libgit2/README.md:6]: we should also mention the last tagged version of the libgit2 package itself, i.e. `git/libgit2/v0.6.0`

**Review by aryan9600 — CHANGES_REQUESTED** [review]

File: git/internal/e2e/run.sh (hunk 2 above)
Comment (stefanprodan) [reply git/internal/e2e/run.sh:20]: We cgo for sha collision detection right?

Comment (aryan9600) [reply git/internal/e2e/run.sh:20]: no we don't need it for sha collision detection but we do need it for `-race` so please ignore this suggestion

**Review by aryan9600 — APPROVED** [review]
lgtm! 🚀

**Pull request merged**